onprc_ehr/resources/scripts/onprc_ehr/onprc_triggers.js

```diff
@@ -1165,7 +1165,110 @@ exports.init = function(EHR){
+
+            }
+            // Ensure users are entering male ids
+            if (row.male )
+            {
+                EHR.Server.Utils.findDemographics({
+                    participant: row.male ,
+                    helper: helper,
+                    scope: this,
+                    callback: function (data)
+                    {
+                        if (data)
+                        {
+                            if (data['gender/origGender'] && data['gender/origGender'] != 'm')
+                                EHR.Server.Utils.addError(scriptErrors, 'male', 'The Male ID has to be male', 'ERROR');
+                        }
+                    }
+                });
+
+            }
+        });
+    });
+
+    // //Added: 10-4-2022  R.Blasa
+    // EHR.Server.TriggerManager.registerHandlerForQuery(EHR.Server.TriggerManager.Events.BEFORE_UPSERT, 'ehr', 'requests', function (helper, scriptErrors, row, oldRow) {
+    //     if (row.requestid && row.qcstate.label ==  'Request: Approved'){
+    //         getRequestCompletedArray: function(){
+    //             var requestid = [];
+    //             for (var i in row.requestid.length){
+    //                 var rows = row.requestid[i];
+    //                 requestid.push(i);
+    //             }
+    //
+    //             return requests;
+    //         }
+    //
+    //
+    //
+    //     }
+    // });
+
+    //Added: 10-4-2022  R.Blasa
+    EHR.Server.TriggerManager.registerHandlerForQuery(EHR.Server.TriggerManager.Events.AFTER_UPSERT, 'ehr', 'requests', function(helper, errors, row, oldRow){
+        if (row.requestid && row.qcstate.label ==  'Request: Approved'){
+
+            var msgs =   helper.getJavaHelper().sendRequestStateEmail("Request: Approved", row.requestid );
+            if (msgs && msgs.length){
+                LABKEY.ExtAdapter.each(msgs, function(msg){
+                    EHR.Server.Utils.addError(scriptErrors, 'qcstate', msg, 'INFO');
+                }, this);
+            }
+
+        }
+    });
+
+    //Added: 10-4-2022  R.Blasa
+    EHR.Server.TriggerManager.registerHandlerForQuery(EHR.Server.TriggerManager.Events.BEFORE_UPSERT, 'ehr', 'requests', function (helper, scriptErrors, row, oldRow) {
+        if (row.requestid && row.qcstate.label ==  'Request: Approved'){
+
+                var requestid = [];
+                for (var i in row.requestid.length){
+                    var rows = row.requestid[i];
+                    requestid.push(i);
+                }
+
+        }
     });
 
+    //Added: 10-4-2022  R.Blasa
+    EHR.Server.TriggerManager.registerHandlerForQuery(EHR.Server.TriggerManager.Events.AFTER_UPSERT, 'ehr', 'requests', function(helper, errors, row, oldRow){
+        if (row.requestid && row.qcstate.label ==  'Request: Approved'){
+            var requestsApproved = triggerHelper.getApprovedArray();
+            var msgs =   helper.getJavaHelper().sendRequestStateEmail("Request: Approved", requestsApproved );
```

Comment:
```suggestion
            var msgs =   helper.getJavaHelper().sendRequestStateEmail("Request: Approved", [row.requestid] );
```
row.requestid should always be a single value in this handler so just making a single value array out of it should work fine.